src/runtime/internal/style_manager.ts

```diff
@@ -18,6 +19,17 @@ function hash(str: string) {
 	return hash >>> 0;
 }
 
+function append_style(doc) {
+	return doc.host ?
+		doc.insertBefore(element('style') as HTMLStyleElement, doc.firstElementChild) :
+		doc.head.appendChild(element('style') as HTMLStyleElement);
+}
+
+function get_root_node(node) {
+	if (is_function(node.getRootNode)) return node.getRootNode();
```

Comment:
I found a weird behaviour doing so, let's suppose we have a transition on hover on an element, if the element on which we want to apply the transition is directly a child of a shadow-root element it will be okay, if otherwise it's a child of a regular DOM node (let's say a div) it will keep on appending `style` tags on each hover. Looking recursively for a parenElement to get to the root of the webComponent solves the issue.

Reply:
`node.getRootNode()` is your friend here.

Reply:
Interesting, thanks. Could you show me your code so I can have a look as well?